rundoctests.py

```diff
@@ -24,47 +65,55 @@
 for typ in [AlarmInterrupt, SignalError]:
     typ.__module__ = "__main__"
 
+if os.name != 'nt':
+    # Limit stack size to avoid errors in stack overflow doctest
+    stacksize = 1 << 20
+    resource.setrlimit(resource.RLIMIT_STACK, (stacksize, stacksize))
+
+    # Disable core dumps
+    resource.setrlimit(resource.RLIMIT_CORE, (0, 0))
+
+def testfile(file):
+    # Child process
+    try:
+        if sys.platform == 'darwin':
+            from cysignals.signals import _setup_alt_stack
+            _setup_alt_stack()
+        failures, _ = doctest.testfile(file, module_relative=False, optionflags=flags, parser=parser)
+        if not failures:
+            os._exit(0)
+    finally:
+        os._exit(23)
+
+if __name__ == "__main__": # Mandatory for windows cases.
+    success = True
+    for f in filenames:
+        print(f)
+        sys.stdout.flush()
+
+        # Test every file in a separate process (like in SageMath) to avoid
+        # side effects from doctests.
+        p = Process(target=testfile, args=(f,))
+        p.start()
+        p.join(timeout)
+
+        status = p.exitcode
```

Comment:
This doesn't work the same way as the return value from `os.waitpid`.

Reply:
It seems that signals are represented by a negative value and that normal exit codes are just passed as-is. So the logic below needs to be fixed.

Reply:
This patch seems to work for me.
```
diff --git a/rundoctests.py b/rundoctests.py
index 4d6bd73..3c5179d 100755
--- a/rundoctests.py
+++ b/rundoctests.py
@@ -106,14 +106,9 @@ if __name__ == "__main__": # Mandatory for windows cases.
         elif status != 0:
             success = False
             if os.name != 'nt':
-                if os.WIFEXITED(status):
-                    st = os.WEXITSTATUS(status)
-                    if st != 23:
-                        print("bad exit: {}".format(st))
-                elif os.WIFSIGNALED(status):
-                    sig = os.WTERMSIG(status)
-                    print("killed by signal: {}".format(sig))
-                else:
-                    print("unknown status: {}".format(status))
+                if status < 0:
+                    print("killed by signal: {}".format(abs(status)))
+                elif status != 23:
+                    print("bad exit: {}".format(status))
```

Reply:
Yes, something like that should work.